src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentDescriptorProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc.ViewComponents
+{
+    /// <summary>
+    /// Default implementation of <see cref="IViewComponentDescriptorProvider"/>/
+    /// </summary>
+    public class DefaultViewComponentDescriptorProvider : IViewComponentDescriptorProvider
+    {
+        private readonly IAssemblyProvider _assemblyProvider;
+
+        /// <summary>
+        /// Creates a new <see cref="DefaultViewComponentDescriptorProvider"/>.
+        /// </summary>
+        /// <param name="assemblyProvider">The <see cref="IAssemblyProvider"/>.</param>
+        public DefaultViewComponentDescriptorProvider(IAssemblyProvider assemblyProvider)
```

Comment:
Whaaa, sure its 99% of the time coming from services but in the end its still a class. When did the engineering team make that decision?